profiles/mojoshader_profile_metal.c

```diff
@@ -558,7 +558,7 @@ void emit_METAL_finalize(Context *ctx)
         output_line(ctx, "};");
         pop_output(ctx);
 
-        output_line(ctx, "constant %s_Uniforms &uniforms [[buffer(16)]]%s", ctx->mainfn, commas ? "," : "");
+        output_line(ctx, "constant %s_Uniforms &uniforms [[buffer(0)]]%s", ctx->mainfn, commas ? "," : "");
```

Comment:
What is this change for again? Let's make this a separate commit, then I'll go ahead and merge the rest; my guess is the remaining Flotilla issues don't have to do with this change as much as the Metal emitter and maybe the pointsize fixups.

Reply:
SDL_GPU requires that buffer indices are set up in a specific order, with uniform buffers appearing first, then vertex buffers, then storage buffers. MojoShader was previously hardcoding MSL uniform buffers to appear at index 16. I believe this was chosen under the assumption that you’d start binding vertex buffers at index 0, with a max of 16 vertex buffer bindings.

However, that assumption does not hold for SDL_GPU. So I changed this so that uniform buffers are now emitted at index 0. This is easy to work around in other engines that may use MojoShader; you just bind vertex buffers starting at index 1 instead. (Or 0 if you’re not using a uniform buffer.) This is what SDL’s Metal driver does internally. More specifically it binds vertex buffers starting at index=`num_uniform_buffers`.

Without this change, SDL pipeline creation will fail due to an incorrect uniform buffer index. Since none of the rest of this PR will actually work without this change, I don’t think we should split this off.

Reply:
Fair enough - we can document this somewhere maybe, in case Ryan is still using this somewhere.